packages/relay/src/lib/errors/JsonRpcError.ts

```diff
@@ -10,8 +10,33 @@ export class JsonRpcError {
 
   constructor(args: { code: number; message: string; data?: string }, requestId?: string) {
     this.code = args.code;
-    this.message = requestId ? `[${constants.REQUEST_ID_STRING}${requestId}] ` + args.message : args.message;
     this.data = args.data;
+
+    const hasRequestId = requestId && !args.message.includes(constants.REQUEST_ID_STRING);
+    this.message = hasRequestId ? `[${constants.REQUEST_ID_STRING}${requestId}] ${args.message}` : args.message;
```

Comment:
Sweet, I definitely like it better this way.

I see that almost all usages of this constructor don't pass the `requestId` argument. There is one in `validators.ts` and the other in `webSocketServer.ts` (this one actually passes `undefined`). So if we want to go one step further, I'd move this logic into `newWithRequestId` and update the call sites.